lib/managers/collaboration-whitelists.js

```diff
@@ -0,0 +1,225 @@
+/**
+ * @fileoverview Manager for the Box Collaboration Whitelist Resource
+ */
+
+'use strict';
+// -----------------------------------------------------------------------------
+// Typedefs
+// -----------------------------------------------------------------------------
+
+/**
+ * Collaboration Whitelist parameter constant
+ * @typedef {string} CollaborationWhitelistDirection Determines the type of restriction for whitelisting for a domain
+ */
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+var urlPath = require('../util/url-path');
+
+
+// ------------------------------------------------------------------------------
+// Private
+// ------------------------------------------------------------------------------
+var BASE_PATH = '/collaboration_whitelist_entries',
+	TARGET_ENTRY_PATH = '/collaboration_whitelist_exempt_targets';
+
+// ------------------------------------------------------------------------------
+// Public
+// ------------------------------------------------------------------------------
+
+/**
+ * Simple manager for interacting with all 'Collaboration Whitelist' endpoints and actions.
+ *
+ * @constructor
+ * @param {BoxClient} client - The Box API Client that is responsible for making calls to the API
+ * @returns {void}
+ */
+function CollaborationWhitelists(client) {
+	this.client = client;
+}
+
+/**
+ * Enum of valid collaboration whitelist directions
+ *
+ * @readonly
+ * @enum {CollaborationWhitelistDirection}
+ */
+CollaborationWhitelists.prototype.collaborationWhitelistDirections = {
+	INBOUND: 'inbound',
+	OUTBOUND: 'outbound',
+	BOTH: 'both'
+};
+
+/**
+ * Used to create a collaboration whitelist.
+ *
+ * API Endpoint: '/collaboration_whitelist_entries'
+ * Method: POST
+ *
+ * @param {string} domain - The domain to be added to the whitelist
+ * @param {CollaborationWhitelistDirection} direction - Inbound refers to collaboration actions within an enterprise. Outbound
+ *                                                      refers to collaboration actions external to an enterprise. Both refers to
+ *                                                      collaboration actions taken within and external to an enterprise
+ * @param {Function} callback - Passed the collaboration whitelist information if it was created successfully
+ * @returns {Promise<Object>} A promise resolve to the collaboration whitelist object
+ */
+CollaborationWhitelists.prototype.create = function(domain, direction, callback) {
```

Comment:
Maybe update this method name to something like `collaborationWhitelist.addDomain`?